optuna/integration/pytorch_distributed.py

```diff
@@ -0,0 +1,249 @@
+from datetime import datetime
+import pickle
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+import warnings
+
+import optuna
+from optuna._experimental import experimental
+from optuna._imports import try_import
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalChoiceType
+
+
+with try_import() as _imports:
+    import torch
+    import torch.distributed as dist
+
+
+@experimental("2.6.0")
+class TorchDistributedTrial(optuna.trial.BaseTrial):
+    """A wrapper of :class:`~optuna.trial.Trial` to incorporate Optuna with PyTorch distributed.
+
+    .. seealso::
+        :class:`~optuna.integration.TorchDistributedTrial` provides the same interface as
+        :class:`~optuna.trial.Trial`. Please refer to :class:`optuna.trial.Trial` for further
+        details.
+
+    See `the example <https://github.com/optuna/optuna/blob/master/
+    examples/pytorch_distributed_simple.py>`__
+    if you want to optimize an objective function that trains neural network
+    written with PyTorch distributed data parallel.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` object or :obj:`None`. Please set trial object in
+            rank-0 node and set :obj:`None` in the other rank node.
+    """
+
+    def __init__(self, trial: Optional[optuna.trial.Trial]) -> None:
+
+        _imports.check()
+        self.delegate = trial
```

Comment:
Because it seems the `delegate` field should be private, how about adding `_` prefix to the name?
```suggestion
        self._delegate = trial
```

Reply:
Thanks! I applied your suggestion in the commit https://github.com/optuna/optuna/pull/2303/commits/9a11d2228ebb2e9955b319b4ee03875e76280617 and https://github.com/optuna/optuna/pull/2303/commits/bb2c2d37ad8f35f1a9cf7ed89dc929a46600ca7d.